src/librustc_typeck/check/generator_interior.rs

```diff
@@ -117,6 +119,20 @@ impl<'a, 'tcx> InteriorVisitor<'a, 'tcx> {
                     unresolved_type, unresolved_type_span
                 );
                 self.prev_unresolved_span = unresolved_type_span;
```

Comment:
Pre-existing, but it seems like it would be clearer to "invert" these nested ifs? In other words, we should group all the unresolved type variable handling into one chunk of code, and the happy path in another.

If you'd rather not do it in this PR then feel free to skip, since it's pre-existing.

Reply:
I'd tackle this in the future.